stan/math/rev/core/var.hpp

```diff
@@ -365,35 +369,41 @@ class var_value {
    * View element of eigen matrices
    * @param i Element to access
    */
-  inline const auto coeff(Eigen::Index i) const {
-    using vari_coeff = decltype((*vi_)(i));
-    using var_coeff = var_value<double>;
-    return var_coeff(new vari_coeff((*vi_)(i)));
+  inline auto coeff(Eigen::Index i) const {
+    using vari_coeff_type = decltype(vi_->coeff(i));
+    auto* vari_coeff = new vari_coeff_type(vi_->coeff(i));
+    reverse_pass_callback([vari_coeff, this, i]() mutable {
+      this->vi_->adj_(i) += vari_coeff->adj_;
+    });
```

Comment:
I don't like having to do a copy for the reverse pass. I guess it works, so we can leave it for now, but I sure am revisiting this soon.

Reply:
Yeah it's a bummer, but I think given the use case it make sense no? Like if most of the time we are using `.coeff()` to assign to a var then I think we need to do something like this right? But I'm fine with it for now so I can go start the plumbing up in the stan repo so we can kick of work on the compiler stuff

Reply:
Can you wait a few days with changes in stan repo and compiler? I have an idea I would like to try on this views. I think we can make them assignable, so there is no need for thius type of matrices to be static. This would allow us to eventuallly completely replace `matrix<var>` with `var<matrix>`.